src/freenet/client/filter/HTMLFilter.java

```diff
@@ -2616,7 +2619,11 @@ void processStyle(HTMLParseContext pc) {
 			String href = getHashString(h, "href");
 			if (href != null) {
 				href = HTMLDecoder.decode(href);
-				href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				if (isIcon) {
+					href = htmlSanitizeURI(href, type, null, null, pc.cb, pc, false);
+				} else {
+					href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				}
```

Comment:
I think you’re right. Changing the charset and maybecharset could result in problems if the code structure changes (for example is fomething comes after the current code).

Are there other content types which need a change in the charset?